cdrs-tokio/src/cluster/send_envelope.rs

```diff
@@ -35,7 +35,11 @@ pub async fn send_envelope<T: CdrsTransport + 'static, CM: ConnectionManager<T>
                         }
                     }
                 },
-                Err(error) => return Some(Err(error)),
+                Err(error) => {
+                    eprintln!("Trasport error: {:?}; for node {:?}", error, node);
+                    node.set_state_to_unknown();
+                    continue 'next_node;
```

Comment:
Jumping to a new node will result in the error being lost.

Reply:
That's the core issue - if we run into a scenario when the node is in fact up, but unreachable by the clients, we'll never know when it becomes available, since there will never be any cluster event about it. In such case we should retry the connection at some point. This, unfortunately, means there are conflicting requirements. I'll find out how the java driver handles such situation.

Reply:
The cpp driver handle it by setting the node temporary down.
It can be down in two ways:
1. setting the node to down when it timeout and every X amount of time, resetting all.
2. Setting the node to down and after X amount of time checking if it still down in different thread.